src/coupled_driver.cpp

```diff
@@ -194,29 +212,19 @@ void CoupledDriver::execute()
   heat.write_step();
 }
 
-void CoupledDriver::compute_temperature_norm(const CoupledDriver::Norm& n,
-                                             double& norm,
-                                             bool& converged)
+double CoupledDriver::temperature_norm(Norm norm)
 {
-  switch (n) {
+  switch (norm) {
   case Norm::L1: {
-    auto n_expr = xt::norm_l1(temperatures_ - temperatures_prev_);
-    norm = n_expr();
-    break;
+    return xt::norm_l1(temperatures_ - temperatures_prev_)();
   }
   case Norm::L2: {
-    auto n_expr = xt::norm_l2(temperatures_ - temperatures_prev_);
-    norm = n_expr();
-    break;
+    return xt::norm_l2(temperatures_ - temperatures_prev_)();
   }
   default: {
-    auto n_expr = xt::norm_linf(temperatures_ - temperatures_prev_);
-    norm = n_expr();
-    break;
+    return xt::norm_linf(temperatures_ - temperatures_prev_)();
```

Comment:
These will fall-through

Reply:
Each block has a `return` statement so control flow can't fall through from one block to another.